vizro-ai/src/vizro_ai/dashboard/utils.py

```diff
@@ -61,3 +63,30 @@ def _register_data(all_df_metadata: AllDfMetadata) -> vm.Dashboard:
 
     for name, metadata in all_df_metadata.all_df_metadata.items():
         data_manager[name] = metadata.df
+
+
+def _dashboard_code(
+    dashboard: vm.Dashboard, extra_callable_defs: Optional[Set[str]] = None, extra_imports: Optional[Set[str]] = None
+) -> str:
+    """Get the code for the dashboard."""
+    return dashboard._to_python(extra_callable_defs=extra_callable_defs, extra_imports=extra_imports)
+
+
+def _process_to_set(list_of_list_of_dict):
```

Comment:
What does this variable look like? It's not at all obvious to me from reading why we end up with a list of list of dicts here. Should we flatten this out before passing it into the function?

It would also be nice to give these variables names that reflect their _use_ rather than their data type also.

Reply:
updated